packages/network-controller/tests/NetworkController.test.ts

```diff
@@ -5716,17 +5716,75 @@ function lookupNetworkTests({
           async ({ controller }) => {
             await setFakeProvider(controller, {
               stubs: [
+                // Called during provider initialization
+                {
+                  request: { method: 'net_version' },
+                  response: { result: '12345' },
+                },
+                // Called via `lookupNetwork` directly
                 {
                   request: { method: 'net_version' },
                   response: { result: '12345' },
                 },
               ],
-              stubLookupNetworkWhileSetting: true,
             });
+            expect(controller.state.networkId).toBe('12345');
 
             await operation(controller);
 
-            await expect(controller.state.networkId).toBe('12345');
+            expect(controller.state.networkId).toBe('12345');
+          },
+        );
+      });
+
+      it('updates the network details', async () => {
+        await withController(
+          {
+            state: initialState,
+          },
+          async ({ controller }) => {
+            await setFakeProvider(controller, {
+              stubs: [
+                // Called during provider initialization
+                {
+                  request: {
+                    method: 'net_version',
+                  },
+                  response: SUCCESSFUL_NET_VERSION_RESPONSE,
+                },
+                {
+                  request: {
+                    method: 'eth_getBlockByNumber',
+                  },
+                  response: {
+                    result: PRE_1559_BLOCK,
+                  },
+                },
+                // Called via `lookupNetwork` directly
+                {
+                  request: {
+                    method: 'net_version',
+                  },
+                  response: SUCCESSFUL_NET_VERSION_RESPONSE,
+                },
+                {
+                  request: {
+                    method: 'eth_getBlockByNumber',
+                  },
+                  response: {
+                    result: POST_1559_BLOCK,
+                  },
+                },
+              ],
+            });
+
+            await operation(controller);
+
+            await expect(controller.state.networkDetails).toStrictEqual({
+              EIPS: {
+                1559: true,
+              },
+            });
```

Comment:
Yeah maybe so. I tried writing one already, but I couldn't get it to fail without these changes. I'm not sure why. I'll try again.

Reply:
Oh I see why it didn't work: this situation doesn't exist.

If the network status was "Available" before this `lookupNetwork` call, then the status wouldn't be changing after a successful call. If the status was something other than "Available" then the network ID would always be `null`, so this condition wouldn't be hit.

For an unsuccessful call this wouldn't be hit either, because the `Promise.all` line would throw.

Effectively this change will only ensure the network details are updated. The network status was already up-to-date in all cases beforehand, weirdly enough.

Reply:
The title and description have been updated to reflect this

Reply:
Ah, okay. That makes sense. Sorry I should have realized that 😅

Reply:
Oh I didn't realize it either! It was a worthwhile discovery.